### PR TITLE
More sim3 ops

### PR DESCRIPTION
**Pull request by tcies**
@schneith ptal. Continuation of #42 .

### CODE REVIEW

**Comment by tcies** [comment]
@schneith bump?

**Comment by tcies** [comment]
@schneith bump? Should I assign to someone else? Or would maybe forking be more comfortable?

File: minkindr/include/kindr/minimal/quat-sim-transform.h [hunk 1]
```diff
@@ -17,20 +19,33 @@ class QuatSimTransformTemplate {
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
   typedef QuatTransformationTemplate<Scalar> Transform;
+  typedef QuatSimTransformTemplate<Scalar> Sim3;
+  typedef Eigen::Matrix<Scalar, 3, 1> Vector;
```
Comment (dymczykm) [inline minkindr/include/kindr/minimal/quat-sim-transform.h]: maybe Vector3?

File: minkindr/include/kindr/minimal/quat-sim-transform.h [hunk 2]
```diff
@@ -17,20 +19,33 @@ class QuatSimTransformTemplate {
...
+  typedef Eigen::Matrix<Scalar, 7, 1> Vector7;
   typedef Eigen::Matrix<Scalar, 3, Eigen::Dynamic> Vectors;
```
Comment (dymczykm) [inline minkindr/include/kindr/minimal/quat-sim-transform.h]: How about Matrix3X to keep consistency with Eigen?

File: minkindr/include/kindr/minimal/quat-sim-transform.h [hunk 3]
```diff
@@ -17,20 +19,33 @@ class QuatSimTransformTemplate {
...
 
-  Eigen::Matrix<Scalar, 4, 4> getTransformationMatrix() const;
-  Transform getTransform() const { return T_A_B_; }
-  Scalar getScale() const { return scale_A_B_; }
+  inline Sim3 inverse() const;
+
+  inline Vector7 log() const;
+
+  inline Eigen::Matrix<Scalar, 4, 4> getTransformationMatrix() const;
+  inline Transform getTransform() const { return T_A_B_; }
```
Comment (dymczykm) [inline minkindr/include/kindr/minimal/quat-sim-transform.h]: would it make sense to return a ref to const here?

File: minkindr/include/kindr/minimal/implementation/quat-sim-transform-inl.h [hunk 4]
```diff
@@ -17,12 +17,46 @@ QuatSimTransformTemplate<Scalar>::QuatSimTransformTemplate(
...
+QuatSimTransformTemplate<Scalar> QuatSimTransformTemplate<Scalar>::operator*(
+    const QuatSimTransformTemplate<Scalar>& rhs) const {
+  // Rl*sl tl         Rr*sr tr         Rl*sl*Rr*sr Rl*sl*tr+tl
+  // 0     1     *    0     1     =    0           1
+  //
+  // -> R = Rl*Rr;  t = Rl*sl*tr+tl = Siml*tr;  s = sl*sr
+  return QuatSimTransformTemplate<Scalar>(
+      QuatTransformationTemplate<Scalar>(
+          T_A_B_.getRotation() * rhs.T_A_B_.getRotation(),
+          operator *(rhs.T_A_B_.getPosition())),
```
Comment (dymczykm) [inline minkindr/include/kindr/minimal/implementation/quat-sim-transform-inl.h]: would it be possible to explictly write out the operation, e.g. ``*this * rhs.T_A_B_.getPos()`` or something like that?

**Comment by tcies** [comment]
all addressed, thanks for the review!

**Comment by tcies** [comment]
@ethzasl-jenkins test this please

**Comment by tcies** [comment]
Weird, Jenkins didn't use the latest commit for Github-based unit test. Might there be some conf problem? Anyways, I manually ran a test on the correct sha1: http://129.132.38.183:8080/job/minkindr/134/

**Comment by tcies** [comment]
Now it does...

**Comment by HannesSommer** [comment]
I must defend Jenkins here : it build `130709fe83b00af1a11793a8157926ca35141c2d` and 
```
commit 130709fe83b00af1a11793a8157926ca35141c2d
Merge: 6ead63c 5ccaf1a
Author: Titus Cieslewski <dev@titus-c.ch>
Date:   Tue Dec 6 14:33:24 2016 +0000

    Merge 5ccaf1a652d831470f810592602b6cb0d3186795 into 6ead63c72ac232918ac7947e4e50a2c1f8cce5d9
```
So in fact it did exactly what it should: build the premerged revision  (the one after merging into the PRs target). Does that explain your surprise?

**Comment by tcies** [comment]
I have no build at 14:33, that must be some other system's time... In the end, yes, it did what it should have done. I guess there was some delay.

**Pull request merged**